tests/pricing_models/hyperdrive_test_calc_in_given_out.py

```diff
@@ -0,0 +1,736 @@
+        # purchased for base.
+        #
+        # 1. out_ = 100; 10% fee; 100k share reserves; 100k bond reserves;
+        #    1 share price; 1 init share price; t_stretch targeting 5% APY;
+        #    6 mo remaining
+        # 2. out_ = 100; 20% fee; 100k share reserves; 100k bond reserves;
+        #    1 share price; 1 init share price; t_stretch targeting 5% APY;
+        #    6 mo remaining
+        # 3. out_ = 10k; 10% fee; 100k share reserves; 100k bond reserves;
+        #    1 share price; 1 init share price; t_stretch targeting 5% APY;
+        #    6 mo remaining
+        # 4. out_ = 80k; 10% fee; 100k share reserves; 100k bond reserves;
+        #    1 share price; 1 init share price; t_stretch targeting 5% APY;
+        #    6 mo remaining
+        # 5. out_ = 200; 10% fee; 100k share reserves; 100k bond reserves;
+        #    2 share price; 1.5 init share price; t_stretch targeting 5% APY;
+        #    6 mo remaining
+        # 6. out_ = 200; 10% fee; 100k share reserves; 1M bond reserves;
+        #    2 share price; 1.5 init share price; t_stretch targeting 5% APY;
+        #    6 mo remaining
+        # 7. out_ = 200; 10% fee; 100k share reserves; 1M bond reserves;
+        #    2 share price; 1.5 init share price; t_stretch targeting 5% APY;
+        #    3 mo remaining
+        # 8. out_ = 200; 10% fee; 100k share reserves; 1M bond reserves;
+        #    2 share price; 1.5 init share price; t_stretch targeting 10% APY;
+        #    3 mo remaining
+        base_in_test_cases = [
+            (  # test 1, token BASE
+                # t_d=0.5
+                # tau=0.02253584403159705
+                # 1-tau=0.977464155968403
+                # t_stretch=22.186877016851916
+                # spot_price=0.9516610350825238
+                TestCaseCalcInGivenOutSuccess(
+                    out=Quantity(amount=100, unit=TokenType.BASE),
+                    market_state=MarketState(
+                        share_reserves=100000,  # base reserves (in share terms) base = share * share_price
+                        bond_reserves=100000,  # PT reserves
+                        share_price=1,  # share price of the LP in the yield source
+                        init_share_price=1,  # original share price pool started
+                    ),
+                    fee_percent=0.1,  # fee percent (normally 10%)
+                    days_remaining=182.5,  # 6.0 months remaining
+                    time_stretch_apy=0.05,  # APY used to calculate time_stretch
+                ),
+                # in_base=97.58374088542769
+                # in_pt=102.54051519598579
+                TestResultCalcInGivenOutSuccess(
+                    without_fee_or_slippage=102.53971546251678,
+                    without_fee=102.54051519598579,
+                    fee=0.25397154625167895,
+                    with_fee=102.79448674223747,
+                ),
+            ),  # end of test 1
```

Comment:
This definitely doesn't need to be fixed in this PR, but I don't like these comments personally. I'm of the opinion that they just add clutter and don't really help to distinguish between the cases. Having said this, I know you're just following the old pattern, so no need to address this here.